modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java

```diff
@@ -306,13 +324,36 @@ public CommerceOrder addCommerceOrder(
 		commerceOrder.setCompanyId(user.getCompanyId());
 		commerceOrder.setUserId(userId);
 		commerceOrder.setUserName(user.getFullName());
-		commerceOrder.setBillingAddressId(billingAddressId);
+
+		if (billingAddressId > 0) {
+			commerceOrder.setBillingAddressId(billingAddressId);
+		}
+		else {
+			_setAccountDefaultBillingAddress(commerceChannel, commerceOrder);
+		}
+
```

Comment:
This check also needs to be after `commerceOrder.setCommerceAccountId(commerceAccountId);`

Line 2681 change introduced does the following check:
```

		CommerceChannelAccountEntryRel commerceChannelAccountEntryRel =
			_commerceChannelAccountEntryRelLocalService.
				fetchCommerceChannelAccountEntryRel(
					commerceOrder.getCommerceAccountId(),
					commerceChannel.getCommerceChannelId(),
					CommerceChannelAccountEntryRelConstants.
						TYPE_BILLING_ADDRESS);			
```
This is always going to fail, otherwise.